README.md

```diff
@@ -1,4 +1,208 @@
+
+**Actions out** are consumed back again by store and trigger calling `reducer` function.
+
+Basic usage for side effects is to perform some job on incoming action, that may take time (I/O operation) 
+or job that should not update the state (same state should be emitted).
+
+Let's add an side effect that makes a network request each time when action is `CalculatorAction.Add`
+and current state is `9`. If server responds with http code `200` - side effect should emit `CalculatorAction.Deduct(1)`.
+```kotlin
+val sideEffect = object : SideEffect<Int, CalculatorAction> {
+    override val name: String = "network logger"
+
+    override fun CoroutineScope.start(
+        input: ReceiveChannel<CalculatorAction>,
+        stateAccessor: StateAccessor<Int>,
+        output: SendChannel<CalculatorAction>,
+        logger: SideEffectLogger
+    ): Job = launch(context = CoroutineName(name)) {
+        for (inputAction in input) {
+            logger.logSideEffectEvent { LogEvent.SideEffectEvent.InputAction(name, inputAction) }
+            if (inputAction is CalculatorAction.Add &&
+                stateAccessor() == 9) {
+                launch {
+                    val response = makeNetworkCall()
+                    logger.logSideEffectEvent {
+                        LogEvent.SideEffectEvent.Custom(name, "Received network response: $response")
+                    }
+                    if (response == 200) {
+                        val outputAction = CalculatorAction.Deduct(1)
+                        logger.logSideEffectEvent { LogEvent.SideEffectEvent.DispatchingToReducer(name, outputAction) }
+                        output.send(outputAction)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Now we need to modify how store is created by adding this `sideEffect`:
+```kotlin
+val storeWithSideEffect = GlobalScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    sideEffects = listOf(sideEffect),
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+When we subscribe to `storeWithSideEffect` and trigger side effect by dispatching right actions - 
+network request from side effect will be performed:
+```kotlin
+storeWithSideEffect.subscribe { state -> updateUI(state) }
+storeWithSideEffect.dispatch(CalculatorAction.Add(9)) 
+// Current state will become 9
+storeWithSideEffect.dispatch(CalculatorAction.Add(1)) 
+// Current state will become 10 and network request will happen
+```
+
+Library provides useful simplified versions of `SideEffect` interface:
```

Comment:
CoRedux  also provides two simplified implementations  of `SideEffect` that you might find useful: